pkg/configuration/base/resources/proxy.go

```diff
@@ -0,0 +1,112 @@
+package resources
+
+import (
+	"context"
+	"fmt"
+
+	apiconfigv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	DefaultOpenShiftProxyNamespace = "openshift-config"
+	envHTTPProxyName               = "HTTP_PROXY"
+	envHTTPSProxyName              = "HTTPS_PROXY"
+	envNoProxyName                 = "NO_PROXY"
+)
+
+var (
+	ProxyAPIAvailable   = false
+	ProxyAPIChecked     = false
+	allProxyEnvVarNames = []string{
+		envHTTPProxyName,
+		envHTTPSProxyName,
+		envNoProxyName,
+	}
+	ProxyEnvVars = []corev1.EnvVar{}
+)
+
+func IsProxyAPIAvailable(clientSet *kubernetes.Clientset) bool {
+	if ProxyAPIChecked {
+		return ProxyAPIAvailable
+	}
+	gv := schema.GroupVersion{
+		Group:   apiconfigv1.GroupName,
+		Version: apiconfigv1.SchemeGroupVersion.Version,
+	}
+	if err := discovery.ServerSupportsVersion(clientSet, gv); err != nil {
+		// error, API not available
+		ProxyAPIChecked = true
+		ProxyAPIAvailable = false
+	} else {
+		// API Exists
+		ProxyAPIChecked = true
+		ProxyAPIAvailable = true
+	}
+	return ProxyAPIAvailable
+}
+
+func QueryProxyConfig(k8sClient client.Client) error {
+	global := &apiconfigv1.Proxy{}
+	err := k8sClient.Get(context.TODO(), types.NamespacedName{Name: DefaultOpenShiftProxyNamespace}, global)
+	if err != nil {
+		logger.Info(fmt.Sprintf("global proxy not found in namespace %s", DefaultOpenShiftProxyNamespace))
+		return err
+	}
+
+	// We have found the global proxy configuration object!
+	ProxyEnvVars = ToEnvVar(global)
+	return nil
+}
+
+func ToEnvVar(proxy *apiconfigv1.Proxy) []corev1.EnvVar {
+	return []corev1.EnvVar{
+		{
+			Name:  envHTTPProxyName,
+			Value: proxy.Status.HTTPProxy,
+		},
+		{
+			Name:  envHTTPSProxyName,
+			Value: proxy.Status.HTTPSProxy,
+		},
+		{
+			Name:  envNoProxyName,
+			Value: proxy.Status.NoProxy,
+		},
+	}
+}
+
+// IsOverridden returns true if the given container overrides proxy env variable(s).
+// We apply the following rule:
+//   If a container already defines any of the proxy env variable then it
+//   overrides all of these.
+func IsProxyOverridden(envVar []corev1.EnvVar) (overrides bool) {
+	for _, envVarName := range allProxyEnvVarNames {
+		_, found := find(envVar, envVarName)
+		if found {
+			overrides = true
+			return
+		}
+	}
+
+	return
+}
+
+func find(proxyEnvVar []corev1.EnvVar, name string) (envVar *corev1.EnvVar, found bool) {
```

Comment:
please rename function to something more specific like getProxyEnvVar or findProxyEnvVar